### DISCUSSION TITLE
How to stop Magit/Emacs from starting a new terminal for every git invocation?

### DISCUSSION THREAD

**Discussion by beatrixada** [magit/magit#4981] [Support] [answered]
I was wondering if it's possible to stop Magit or Emacs from starting a new terminal for every git invocation. My use case is fairly eccentric and involves using the 1Password CLI to authenticate git over https: https://developer.1password.com/docs/cli/shell-plugins/github. The CLI remembers biometric authentication per-terminal session, but I have to reauthorize for every remote operation when using git from Magit. I assume this is because the way Magit calls git (appears to be Emacs `start-file-process`: https://github.com/magit/magit/blob/93d30c5459e75a0d53581da2a5539a54efa329ab/lisp/magit-process.el#L588) is like opening a new terminal for every invocation from 1Password's perspective.

Is there any way for me to change this behavior in my personal Emacs to make Emacs or Magit reuse one terminal for git commands, or submit a PR to add this as a configuration? I'm not expecting other folks to work on this since it's such an idiosyncratic problem.

**Comment by tarsius** [comment, marked as answer]
Magit does not use a terminal to run `git`. Doing that would complicate things a lot and would be highly unidiomatic.

A better approach would probably be to figure out how that 1password tool works. I would expect that it can be made to work without using a terminal. Maybe all you have to do is set some environment variables.

**Reply by kyleam** [reply]
And it might be worth checking out https://github.com/xuchunyang/1password.el or other libraries that work with 1password.

**Reply by beatrixada** [reply]
I’ll do some more digging on the 1Password side, I’m not quite sure how it identifies terminal sessions. To be clear, it does work with Magit, it just requires a reauthorization for every operation requiring credentials, whereas using Git from the terminal it seems to cache the authorization for a bit for each session.
